Initializing a project does not work with 3.3.0

### What broke? What's expected?

I don't know if it is a corner-case related to some of my config, or some kind of PEBCAK because it seems so big, 
but when I try to initialize a new project it simply fails for me (cf *Reproducing this issue*). 

### Reproducing this issue

```
me@t490:~$ mkdir -p ~/projects/guestbook
me@t490:~$ cd ~/projects/guestbook
me@t490:~$ kubebuilder init --domain my.domain --repo my.domain/guestbook
Writing kustomize manifests for you to edit...
Writing scaffold for you to edit...
Get controller runtime:
$ go get sigs.k8s.io/controller-runtime@v0.11.0
# sigs.k8s.io/json/internal/golang/encoding/json
../../go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
../../go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
Error: failed to initialize project: unable to scaffold with "base.go.kubebuilder.io/v3": exit status 2
```

### KubeBuilder (CLI) Version

Version: main.version{KubeBuilderVersion:"3.3.0", KubernetesVendor:"1.23.1", GitCommit:"47859bf2ebf96a64db69a2f7074ffdec7f15c1ec", BuildDate:"2022-01-18T17:03:29Z", GoOs:"linux", GoArch:"amd64"}

### PROJECT version

The project is not scaffolded yet.

### Plugin versions

```yaml
The project is not scaffolded yet.
```


### Other versions

me@t490:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 21.10
Release:	21.10
Codename:	impish

me@t490:~$ go version
go version go1.16.12 linux/amd64

me@t490:~$ kubectl version --client
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:41:01Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}



### Extra Labels

_No response_

--- a/docs/book/src/quick-start.md
+++ b/docs/book/src/quick-start.md
@@ -9,8 +9,9 @@ This Quick Start guide will cover:
 
 ## Prerequisites
 
-- [go](https://golang.org/dl/) version v1.15+ (kubebuilder v3.0).
-- [go](https://golang.org/dl/) version v1.16+ (kubebuilder v3.1+).
+- [go](https://golang.org/dl/) version v1.15+ (kubebuilder v3.0 < v3.1).
+- [go](https://golang.org/dl/) version v1.16+ (kubebuilder v3.1 < v3.3).
+- [go](https://golang.org/dl/) version v1.17+ (kubebuilder v3.3+).
 - [docker](https://docs.docker.com/install/) version 17.03+.
 - [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) version v1.11.3+.
 - Access to a Kubernetes v1.11.3+ cluster.
@@ -92,7 +93,7 @@ and the `controllers/guestbook_controller.go` where the reconciliation business 
 logic. For more info see [Designing an API](/cronjob-tutorial/api-design.md) and [What's in
 a Controller](cronjob-tutorial/controller-overview.md).
 
-If you are editing the API definitions, generate the manifests such as CRs or CRDs using 
+If you are editing the API definitions, generate the manifests such as CRs or CRDs using
 ```bash
 make manifests
 ```
@@ -201,13 +202,13 @@ make deploy IMG=<some-registry>/<project-name>:tag
 <aside class="note">
 <h1>registry permission</h1>
 
-This image ought to be published in the personal registry you specified. And it is required to have access to pull the image from the working environment. 
+This image ought to be published in the personal registry you specified. And it is required to have access to pull the image from the working environment.
 Make sure you have the proper permission to the registry if the above commands don't work.
 
 <h1>RBAC errors</h1>
 
 If you encounter RBAC errors, you may need to grant yourself cluster-admin
-privileges or be logged in as admin. See [Prerequisites for using Kubernetes RBAC on GKE cluster v1.11.x and older][pre-rbc-gke] which may be your case.  
+privileges or be logged in as admin. See [Prerequisites for using Kubernetes RBAC on GKE cluster v1.11.x and older][pre-rbc-gke] which may be your case.
 
 </aside>
 

